server/server.go

```diff
@@ -289,9 +294,19 @@ func (s *server) invokeHandleEndpoint() endpoint.Endpoint {
 	return func(ctx context.Context, args, resp interface{}) (err error) {
 		ri := rpcinfo.GetRPCInfo(ctx)
 		methodName := ri.Invocation().MethodName()
-		if methodName == "" && s.svcInfo.ServiceName != serviceinfo.GenericService {
+		serviceName := ri.Invocation().ServiceName()
+		svc := s.svcMap[serviceName]
+		if svc == nil {
+			return fmt.Errorf("service[%s] is not registered, should not happen", serviceName)
+		}
+		svcInfo := svc.GetServiceInfo()
+		handler := svc.GetHandler()
+		if methodName == "" && svcInfo.ServiceName != serviceinfo.GenericService {
 			return errors.New("method name is empty in rpcinfo, should not happen")
 		}
+		if handler == nil || reflect.ValueOf(handler).IsNil() {
```

Comment:
I think handler and svcInfo of a non nil svc should also be non nil, and this shoud be a default convention. Otherwise judge nil everywhere is awkward.